infra/src/api/io_services_cms/v1/createservice_policy/policy.xml

```diff
@@ -6,6 +6,30 @@
             (because every subscription creation modifies the arm resource).
             In order to mitigate so, we limit concurrency ourself.-->
         <limit-concurrency key="constant-key-all-users" max-count="10" />
+        <choose>
+            <when condition="@(context.Request.Body != null)">
+                <set-body>
+@{
+  var stringBody = context.Request.Body.As<string>();
+  try {
+    var jsonBody = JObject.Parse(stringBody);
+    try {
+      if (context.Request.Headers.ContainsKey("x-user-groups-selc")) {
+        jsonBody["data"]["groupId"] = context.Request.Headers["x-user-groups-selc"][0];
+      } else {
+        jsonBody["data"]["groupId"].Parent.Remove();
```

Comment:
Did you try this? I honestly don't remember if `Remove` is a void method or returns a reference to the existing collection without the removed data

Reply:
I've tested both policy using the APIM "echo API" 😉